src/charm.py

```diff
@@ -407,11 +411,30 @@ def _gen_certs(self):
 
         return ret_certs
 
+    def _process_gateway_info(self):
+        """Parse gateway information and store it, if it is not empty."""
+        try:
+            gateway_info = self._gateway.get_relation_data()
+        except GatewayRelationError:
+            self.model.unit.status = WaitingStatus("Waiting for gateway info relation")
+            return
+        if gateway_info["gateway_namespace"] and gateway_info["gateway_name"]:
+            self._gateway_info["namespace"] = gateway_info["gateway_namespace"]
+            self._gateway_info["name"] = gateway_info["gateway_name"]
+
+    def _get_istio_gateway(self):
+        """Parse 'gateway_info' and return Istio gateway definition in appropriate format."""
+        istio_gateway = ""
+        if self._gateway_info and self._gateway_info["namespace"] and self._gateway_info["name"]:
+            istio_gateway = self._gateway_info["namespace"] + "/" + self._gateway_info["name"]
+        return istio_gateway
```

Comment:
This method does return an istio gateway, but I wonder if "get istio gateway" sends the wrong message.  Since this charm does commonly fetch data from other places like relations, *get*_istio_gateway on first read made me think this would retrieve something.  Since we're using data we've already retrieved, what about calling this just `_istio_gateway` or something similar?

Reply:
Agree here. `get` implies some action. This would make more like it a property.
I was struggling here, because what this method does is prepares Istio gateway in specific format from `gateway_info` that might not even be Istio... anyway, `_istio_gateway` is better in this case.

Reply:
yeah I like where you've gone with it now

Reply:
I just realised that if gateway is not available we would return `None` from `_get_isitio_gateway()` and that would be assigned to environment variable `ISTIO_GATEWAY`:
```
+            "ISTIO_GATEWAY": self._get_istio_gateway(),
```